### PR TITLE
fix: updating virtualNetwork API for LZ network resource

### LINKED ISSUE #3330: [AVM Module Issue]: avm/ptn/lz/sub-vending: Re-running tries to delete a subnet (and possibly more)
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

I'm not sure

### Module Name

avm/ptn/lz/sub-vending

### (Optional) Module Version

0.1.2

### Description

I've got an Azure subscription vending repository using Bicep and Azure DevOps pipelines to manage the creation, update and cancelling of Azure subscriptions.

I'm using the `avm/ptn/lz/sub-vending` module to handle the subscription creation/update part of the process.

I've created a subscription using the module, which went fine.

Another team has then proceeded to populate the VNet with subnets and other resources, as expected. One of these resources is an AKS cluster.

I've since re-run the pipeline to update the same subscription by enabling more resource providers in the `avm/ptn/lz/sub-vending` module configuration and it fails trying to delete one of the subnets.

The bottom-most error message in a nested stack of messages is this:
```json
{
    "code": "InUseSubnetCannotBeDeleted",
    "message": "Subnet alz-spoke-t2awudh7kzl5o-eastus2-subnet-AKS is in use by /subscriptions/xxx/resourceGroups/MC_rg-alz-landingzones-online_xxx-akscluster_eastus2/providers/Microsoft.Network/networkInterfaces/|providers|Microsoft.Compute|virtualMachineScaleSets|aks-agentpool-xxx-vmss|virtualMachines|0|networkInterfaces|aks-agentpool-xxx-vmss/ipConfigurations/ipconfig1 and cannot be deleted. In order to delete the subnet, delete all the resources within the subnet. See aka.ms/deletesubnet.",
    "details": []
}
```

Is this expected behaviour?

I thought Bicep ran in incremental mode by default, not complete mode, so it shouldn't be attempting to delete anything.

I need the `avm/ptn/lz/sub-vending` module to be able to create and *update* subscriptions when needed, such that it only updates the configuration/resources that it owns. I don't want it to interfere with resources created outside of it.

### (Optional) Correlation Id

_No response_

### PR DESCRIPTION
**Pull request by sebassem**
## Description
Bump the LZ virtual network resource to avoid subnet deletion on PUT requests
Fixes #3330 

## Pipeline Reference

| Pipeline |
| -------- |
|  [avm.ptn.lz.sub-vending](https://github.com/sebassem/bicep-registry-modules/actions/workflows/avm.ptn.lz.sub-vending.yml)        |

## Type of Change

- [ ] Update to CI Environment or utilities (Non-module affecting changes)
- [ ] Azure Verified Module updates:
  - [X] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [X] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [X] I'm sure there are no other open Pull Requests for the same update/change
- [X] I have run `Set-AVMModule` locally to generate the supporting module files.
- [X] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

File: avm/res/network/virtual-network-gateway/main.json
Comment (eriqua) [inline avm/res/network/virtual-network-gateway/main.json:1]: Please remove this change if unrelated

Comment (sebassem) [reply avm/res/network/virtual-network-gateway/main.json:1]: Resolved

**Review by eriqua — APPROVED** [review]
🆗

**Comment by eriqua** [comment]
Hey @sebassem the PR is approved. Feel free to merge

**Pull request merged**